### ISSUE TITLE
ActorFuture roadmap

### ISSUE THREAD

**Issue by PidgeyBE** [dask/distributed#2950]
Hi!

Currently `ActorFuture`s are not really Futures, but just a caching mechanism.
As explained here: https://distributed.dask.org/en/latest/actors.html#call-remote-methods,
> They can’t be placed into additional submit or map calls to form dependencies. 
For my use case, I want to have a dag, in which every part of the dag is persistent. So I need to pass the output of 1 part of the dag (=`ActorFuture`) as a parameter to the submit call of the next part. Unfortunately this is not possible yet. (I know in similar projects like Ray, this is possible).
I was wondering if someone knows what the plans with Dask Actor(Future)s are?
Will ActorFuture be fully featured Futures soon or..?

Thanks for helping! Pj

**Comment by mrocklin** [comment]
There is no roadmap or planned work for this as far as I know.  People tend
to work on what they need, rather than following a prescribed plan.

**Comment by TomAugspurger** [comment]
@PidgeyBE are you interested in working on this? Either way, can you provide a short code snippet demonstrating the issue?

Matt can correct me if I'm wrong, but it seems like it would be in scope since it fits with our handling of Future objects.

**Comment by mrocklin** [comment]
Yes, this work seems in scope.  It's challenging because the scheduler
doesn't keep track of ActorFutures.  ActorFutures are intentionally not
tracked by the scheduler because there can be many more of them (billions),
so there would need to be some way to promote an ActorFuture into a normal
future that didn't affect the common case.

**Comment by TomAugspurger** [comment]
I'm a little concerned about how this will interact with the `pure` keyword and out caching... I can pretty easily see someone forgetting to set `pure=False` and getting "stale" results because the actor's value changed. Is that worry unreasonable?

**Comment by mrocklin** [comment]
An ActorFuture often points to a non-changing result.  These aren't futures that point to Actors, these are pointers to eventual results of computations by actors.

**Comment by jakirkham** [comment]
Any thoughts on this thus far, @PidgeyBE? Are you interested in pursuing this work?

**Comment by PidgeyBE** [comment]
@jakirkham @TomAugspurger I'm sorry but I've decided continuing my project with Ray, since Ray already supports "real" ActorFutures and additionally it supports multiple outputs for a remotely executed function. (e.g. 1 function can return multiple Futures instead of just one, which is handy if you want to construct complex graphs)

**Comment by jherreral** [comment]
I recently worked on a distributed rendering project using actors, and though I managed to get decent results it was much more effort than it should be. The first next piece of API that I'd like to see is a status attribute in the ActorFuture that allows checking if some remotely called method has hanged or not.
Are there any plans to tackle this?

**Comment by TomAugspurger** [comment]
Not that I know of. Are you interested in working on it @jherreral?
